scripts/irods/test/test_auth.py

```diff
@@ -99,85 +99,93 @@ def test_authentication_PAM_without_negotiation(self):
+                                'password_length': 20,
+                                'no_extend': False,
+                                'password_min_time': 121,
+                                'password_max_time': 1209600,
+                                }
                             }
                         }
-                    }
-                lib.update_json_file_from_dict(irods_config.server_config_path, server_config_update)
+                    lib.update_json_file_from_dict(irods_config.server_config_path, server_config_update)
 
-                pep_map = {
-                    'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
-                        acPreConnect(*OUT) {
-                            *OUT = 'CS_NEG_REQUIRE';
-                        }
-                    '''),
-                    'irods_rule_engine_plugin-python': textwrap.dedent('''
-                        def acPreConnect(rule_args, callback, rei):
-                            rule_args[0] = 'CS_NEG_REQUIRE'
-                    ''')
-                }
+                    pep_map = {
+                        'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
+                            acPreConnect(*OUT) {
+                                *OUT = 'CS_NEG_REQUIRE';
+                            }
+                        '''),
+                        'irods_rule_engine_plugin-python': textwrap.dedent('''
+                            def acPreConnect(rule_args, callback, rei):
+                                rule_args[0] = 'CS_NEG_REQUIRE'
+                        ''')
+                    }
 
-                with temporary_core_file() as core:
-                    time.sleep(1)  # remove once file hash fix is committed #2279
-                    core.add_rule(pep_map[self.plugin_name])
-                    time.sleep(1)  # remove once file hash fix is committed #2279
+                    with temporary_core_file() as core:
+                        time.sleep(1)  # remove once file hash fix is committed #2279
+                        core.add_rule(pep_map[self.plugin_name])
+                        time.sleep(1)  # remove once file hash fix is committed #2279
 
-                    IrodsController().start()
+                        IrodsController().start()
 
-                    # the test
-                    print(f'running iinit for PAM user [{self.auth_session.username}] [{self.auth_session.password}]')
-                    self.auth_session.assert_icommand('iinit', 'STDOUT_SINGLELINE',
-                                                      input=f'{self.auth_session.password}\n')
-                    self.auth_session.assert_icommand("icd")
-                    self.auth_session.assert_icommand("ils -L", 'STDOUT_SINGLELINE', "home")
+                        print('attempting to run an icommand')
+                        lib.execute_command(['ils'])
+                        print('Success!!')
```

Comment:
Can you explain why this is necessary?

Just curious as to why `lib.execute_command` is used. It the `ils` fails, would that be caught or mean anything regarding the test?

Reply:
Oh, this is leftover debugging code. I will remove this.

For posterity, `lib.execute_command` is used instead of a session because the client environment for the service account is being modified in this test. I wanted to confirm that it wasn't getting screwed up before attempting to run something as some other account. It's not really any more valid than using one of the sessions, but removed a layer in my debugging journey.

Reply:
Understood.